data/cluster_id.go

```diff
@@ -53,16 +51,7 @@ func NewClusterIDFromPersistentStorage(sgc KubeSecretGetterCreator) ClusterID {
 func (c clusterIDFromPersistentStorage) Get() (string, error) {
 	secret, err := c.secretGetterCreator.Get(wfmSecretName)
 	if err != nil {
-		log.Printf("Error getting secret [%s]", err)
-		switch e := err.(type) {
-		case *k8sErrors.StatusError:
-			// If the error isn't a 404, we don't know how to deal with it
-			if e.ErrStatus.Code != 404 {
-				return "", err
-			}
-		default:
-			return "", err
-		}
+		return "", err
```

Comment:
so the idea now is that if k8s returns a 404, it's still an error from `Get()`'s perspective?